### PR TITLE
Add get pending transaction method for multisig

### PR DESCRIPTION
**Pull request by sudo-shashank**
**Summary of changes**
Changes introduced in this pull request:
- Added `get_pending_txn` method required for `MsigGetPending` [rpc](https://github.com/ChainSafe/forest/pull/3838)
- Use `fvm_shared4` instead of `fvm_shared` for multisig v12 

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: fil_actor_interface/src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#[macro_export]
+macro_rules! parse_pending_transactions {
```
Comment (LesnyRumcajs) [inline fil_actor_interface/src/macros.rs:5]: It'd be nice to have some documentation here.

File: fil_actor_interface/src/builtin/multisig/mod.rs [hunk 2]
```diff
@@ -84,7 +99,71 @@ impl State {
...
+                        value: from_token_v4_to_v2(txn.value.clone()),
+                        method: txn.method,
+                        params: txn.params.clone(),
+                        approved: txn
+                            .approved
+                            .iter()
+                            .map(|&addr| from_address_v4_to_v2(addr))
+                            .collect(),
+                    });
+                    Ok(())
```
Comment (LesnyRumcajs) [inline fil_actor_interface/src/builtin/multisig/mod.rs:163]: Could this be hidden behind a macro/function to keep it consistent with the variants above?

File: fil_actor_interface/src/macros.rs (hunk 1 above)
Comment (sudo-shashank) [reply fil_actor_interface/src/macros.rs:26]: added

File: fil_actor_interface/src/builtin/multisig/mod.rs (hunk 2 above)
Comment (sudo-shashank) [reply fil_actor_interface/src/builtin/multisig/mod.rs]: done

File: fil_actor_interface/src/macros.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+/// `parse_pending_transactions` is a macro for parsing pending transactions and populating a vector with transaction data.
```
Comment (LesnyRumcajs) [inline fil_actor_interface/src/macros.rs:4]: This does a bit too much, several optional arguments make it difficult to analyze. Did you consider splitting it into smaller methods?

Comment (sudo-shashank) [reply fil_actor_interface/src/macros.rs:4]: yes initially i tried to have it behind one macro only, but looking it now the no of args does seems a lot to manage. 
I think having smaller methods here is a better way to go

Comment (LesnyRumcajs) [reply fil_actor_interface/src/macros.rs:4]: Let's split it, then.

**Review by LesnyRumcajs — APPROVED** [review]

**Review by jdjaustin — APPROVED** [review]

**Pull request merged**